profiles/permissions_test.py

```diff
@@ -47,6 +59,127 @@ def test_cant_edit_if_not_owner(self):
         other_user = UserFactory.create()
         for method in ('POST', 'PATCH', 'PUT'):
             with mute_signals(post_save):
-                profile = ProfileFactory.create()
+                profile = ProfileFactory.create(account_privacy=Profile.PUBLIC_TO_MM)
+
             request = Mock(method=method, user=other_user)
             assert not perm.has_object_permission(request, None, profile)
+
+
+# pylint: disable=no-self-use
+class CanSeeIfNotPrivateTests(TestCase):
+    """
+    Tests for CanEditIfPrivate permissions
+    """
+
+    def setUp(self):
+        super(CanSeeIfNotPrivateTests, self).setUp()
+        with mute_signals(post_save):
+            self.other_user = other_user = UserFactory.create(username="test")
+            username = "{}_edx".format(other_user.username)
+            other_user.social_auth.create(
+                provider=EdxOrgOAuth2.name,
+                uid=username
+            )
+            self.other_user_profile = ProfileFactory.create(user=other_user, verified_micromaster_user=False)
+
+    def test_cant_view_if_privacy_is_private(self):
+        """
+        Users are not supposed to view private profiles.
+        """
+        perm = CanSeeIfNotPrivate()
+
+        for method in ('POST', 'PATCH', 'PUT'):
+            with mute_signals(post_save):
+                profile_private = ProfileFactory.create(account_privacy=Profile.PRIVATE)
+
+            request = Mock(method=method, user=self.other_user)
+            result = Mock(method='get_object_or_404', return_value=profile_private)
```

Comment:
`method='get_object_or_404'` doesn't have any meaning here. Since you are mocking a function only `return_value` matters